tests/test_routes.py

```diff
@@ -0,0 +1,91 @@
+from flask import session
+from urlshortener.extensions import mongo
+
+
+def test_home(client):
+    response = client.get("/")
+
+    assert response.status_code == 200
+    assert b"<title>URL Shortener</title>" in response.data
+
+
+def test_add(client):
+    client.post("/add", data={"long_url": "https://www.url.test"})
+
+    assert mongo.db.links.estimated_document_count() == 1
+    assert len(mongo.db.links.find_one()["short_url"]) == 3
+    assert mongo.db.links.find_one()["long_url"] == "https://www.url.test"
+
+
+def test_short_url_and_info(client):
+    client.post("/add", data={"long_url": "https://www.url1.test"})
+    client.post("/add", data={"long_url": "https://www.url2.test"})
+
+    link1 = mongo.db.links.find_one({"long_url": "https://www.url1.test"})
+    link2 = mongo.db.links.find_one({"long_url": "https://www.url2.test"})
+
+    mongo.db.links.update_one(
+        {"_id": link1["_id"]},
+        {"$set": {"short_url": "add"}})
+    mongo.db.links.update_one(
+        {"_id": link2["_id"]},
+        {"$set": {"short_url": "short1"}})
```

Comment:
Seeing these manuevers and the comment above, I think you want to refactor your app so that you can inject a function that returns predictable short codes for testing.